_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,285 @@
+---
+title: "Introduction to Unit Testing using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What framework/libraries can be used for unit testing in C++?"
+- "What is GoogleTest?"
+- "How to write unit tests with GoogleTest?"
+objectives:
+- "Understand the basic components of a unit test"
+- "Understand how to write a unit test using GoogleTest"
+- "Understand how to compile and run the tests"
+- "Understand some of the macros available in GoogleTest"
+keypoints:
+- "We learnt about some of the popular testing frameworks for C++."
+- "We learnt the baiscs of GoogleTest."
+- "We learnt how to use GoogleTest to write our tests."
+- "We learnt about anatomy of a unit tests." 
+- "We learnt how to write effective unit tests."
+- "Finally, we saw an example to demonstrate the above concepts."
+---
+
+## 1. Popular frameworks for testing in C++
+In C++, a framework refers to a pre-defined software structure or set of libraries that provides a foundation for developing applications or solving specific types of problems. It typically includes a collection of reusable code, tools, and guidelines that help developers build applications more efficiently. There can be different types of frameworks in C++ which include:
+
+1. Application Framework
+2. Web Framework
+3. Game Development Framework
+4. Testing Framework
```

Comment:
On second reading, I'm not sure we need to explain what a framework is. Yes, it is being used with a somewhat technical meaning here, but I think it's pretty obvious from the context.

I think the reader would understand fine if you just jumped straight into explaining what a testing framework is.

Reply:
@alexdewar I did not introduce as it should have been obvious. It was @dalonsoa suggestion and then I added it.

Reply:
@dalonsoa Not sure we need this... Even with not knowing the technical meaning of framework, I think it would be easy to follow. What do you reckon?

Reply:
Resolved in 9872529e65183b3cf8fe16120bf43ef5f4658a72